test_cli.py

```diff
@@ -1195,6 +1201,51 @@ def test(ctx):
     )  # 2 header lines, 1 stdout file, 1 stderr file, 1 item at bundle target root
 
 
+@TestModule.register('link')
+def test(ctx):
+    # Upload fails
+    uuid = _run_command([cl, "upload", "/etc/passwd", '--link'])
+    check_equals(State.READY, get_info(uuid, 'state'))
+    _run_command([cl, 'cat', uuid], 1)
+
+    # Upload file
+    # We create the file at /opt/codalab-worksheets-link-mounts/tmp because this test is running
+    # inside a Docker container (so the host directory /tmp is mounted at /opt/codalab-worksheets-link-mounts/tmp).
+
+    with tempfile.NamedTemporaryFile(
+        mode='w', dir='/opt/codalab-worksheets-link-mounts/tmp', suffix=".txt", delete=False
+    ) as f:
+        f.write("hello world!")
```

Comment:
nit: can we end with a new line (use print)?

Reply:
Wouldn't using print to a file make this far more complicated (because it requires redirecting stdout)? see https://stackabuse.com/writing-to-a-file-with-pythons-print-function/